### PR TITLE
[ARM] Fix az resource tag crashed when the parameter --ids passed in is resource group ID

### PR DESCRIPTION
**Pull request by zhoxing-ms**
Fix `az resource tag` crashed when the parameter `--ids` passed in is resource group ID
Issue: #10246

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/resource/custom.py [hunk 1]
```diff
@@ -2268,6 +2270,8 @@ def resolve_api_version(rcf, resource_provider_namespace, parent_resource_path,
     @staticmethod
     def _resolve_api_version_by_id(rcf, resource_id):
         parts = parse_resource_id(resource_id)
+        if 'namespace' not in parts:
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/resource/custom.py]: use more detailed condition for resource group

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/custom.py]: Done

File: src/azure-cli/azure/cli/command_modules/resource/custom.py [hunk 2]
```diff
@@ -2268,6 +2268,13 @@ def resolve_api_version(rcf, resource_provider_namespace, parent_resource_path,
     @staticmethod
     def _resolve_api_version_by_id(rcf, resource_id):
         parts = parse_resource_id(resource_id)
+
+        if len(parts) == 2 and parts['subscription'] is not None and parts['resource_group'] is not None:
+            return AZURE_API_PROFILES['latest'][ResourceType.MGMT_RESOURCE_RESOURCES]
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/resource/custom.py]: I'd prefer to add multi-API support in `azure.cli.command_modules.resource.custom._ResourceUtils._resolve_api_version_by_id` for resource group operations and use corresponding API versions for `ResourceType.MGMT_RESOURCE_RESOURCES`.

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/custom.py]: `az resource tag` has the `--api-version` parameter which provider the api version of the resource (omit for latest version).

File: src/azure-cli/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+**ARM**
+
+* Fix issue #10246: `az resource tag` crashed when the parameter `--ids` passed in is resource group ID
```
Comment (jiasli) [inline src/azure-cli/HISTORY.rst]: Please use present tense “crash" -> "crashes".

Comment (zhoxing-ms) [reply src/azure-cli/HISTORY.rst]: Done

**Comment by zhoxing-ms** [comment]
Reopen and merge

**Pull request merged**